core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -400,7 +398,9 @@ private[kafka] class Processor(val id: Int,
 
   private val newConnections = new ConcurrentLinkedQueue[SocketChannel]()
   private val inflightResponses = mutable.Map[String, RequestChannel.Response]()
-  private val metricTags = Map("networkProcessor" -> id.toString).asJava
+  val metricTags = Map(
+    "protocol" -> securityProtocol.name, "listener" -> listenerName.value, "networkProcessor" -> id.toString
+    ).asJava
```

Comment:
Looking at the JMX screenshot, the order is important, so you should use a `LinkedHashMap`. Also, it seems like `listener` should be the first entry as it uniquely identifies the listener.